File: next-dashboard-ui/src/lib/lesson.ts
import { Subject } from "./subject";
import { Teacher } from "./teacher";
import { Class } from "./class";

export interface Lesson {
  lessonID: number;
  lessonName?: string;
  lessonDate?: string;

  subjectID?: number;
  subject?: Subject;

  teacherID?: number;
  teacher?: Teacher;

  classID?: number;
  class?: Class;
}

File: backend-asp-school/backend-asp-school/Models/Lesson.cs
using System.Security.Claims;
using System.Text.Json.Serialization;

namespace backend_asp_school.Models
{
    public class Lesson
    {
        public int LessonID { get; set; }
        public string? LessonName { get; set; }
        public DateTime? LessonDate { get; set; }

        public int? SubjectID { get; set; }
      
        public Subject? Subject { get; set; }

        public int? TeacherID { get; set; }
       
        public Teacher? Teacher { get; set; }

        public int? ClassID { get; set; }
       
        public Classes? Class { get; set; }
    }
}
